ctera/src/main/java/ch/cyberduck/core/ctera/CteraSession.java

```diff
@@ -75,23 +91,68 @@
 import java.text.MessageFormat;
 import java.time.Duration;
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import com.github.sardine.DavResource;
 import com.google.common.util.concurrent.Uninterruptibles;
 
+import static ch.cyberduck.core.ctera.CteraAclPermissionFeature.*;
+
+
 public class CteraSession extends DAVSession {
+    // TODO CTERA-136 create Ctera***Feature for all preflight cases? Keeping all checks in one location (CteraSession) for now.
```

Comment:
Add features instead as the preflight checks can be dependent on the feature context.

Reply:
Done e57ba170367804bdddabca3d79878d9cc6a410ec